Fixes and updates for i18n tools

rtibbles opened the pull request:
## Summary
* Fixes long standing bug in the fonts.py that failed to defer import of notosource until after fontTools is installed
* Adds message extraction from the kolibri package
* Updates module resolution to use the webpack enhanced-resolver to allow us to continue to effectively resolve imports with dynamic aliases when needed
* Add tests for module resolution
* Add tests for a range of AST utilities to try to make them less brittle in future.
* Adds import name resolution for `require` statements and dynamic `import` statements.
* Adds tests.
* Updates committed message files.

Fixes https://github.com/learningequality/kolibri/issues/13314

## Reviewer guidance
Running `make i18n-download` and ensuring it gives you no diff is probably the easiest thing here.

rtibbles committed:
Delay notosource import until we have installed fontTools.

rtibbles committed:
Ensure we extract message from the kolibri package.

rtibbles committed:
Update string extraction to use webpack's enhanced resolver.
Add covering tests for module resolution and AST parsing utilities.

marcellamaki commented on packages/kolibri-tools/lib/i18n/utils.js:
```diff
@@ -16,7 +16,7 @@ function checkForDuplicateIds(obj1, obj2) {
     const message1 = obj1[potentialDuplicate].message;
     const message2 = obj2[potentialDuplicate].message;
-    if (message1 !== message1) {
+    if (message1 !== message2) {
```
this is an important difference

marcellamaki commented on packages/kolibri-tools/lib/i18n/__tests__/astUtils.spec.js:
```diff
@@ -0,0 +1,480 @@
+      'node-module-import.js',
+      `
+      // A file with an import from node_modules
+      import React from 'react';
```
is the fact that this is react not vue meaningful? like, because we don't use it?

rtibbles replied:
No - this is just what happens when I only skim the contents of the test files that Claude creates.

rtibbles committed:
Properly handle require and dynamic import statements.

rtibbles committed:
Incremental translation update.
Restore previously missing kolibri package messages that were not put into message files.

marcellamaki reviewed (approved):
the code changes lgtm - and running make i18n-download seems to be working well, the checks/info messages make sense and print in the console, and the diff accurately reflects the correct changes after spot checking - new strings related to the feature as well as messages in `core`
